projects/sound-open-firmware/Dockerfile

```diff
@@ -20,39 +20,57 @@
 
 FROM gcr.io/oss-fuzz-base/base-builder
 
-RUN apt-get -y update && \
-	    apt-get install -y \
-	    	autoconf \
-		bison \
-		build-essential \
-		flex \
-		gawk \
-		gettext \
-		git \
-		gperf \
-		help2man \
-		libncurses5-dev \
-		libssl-dev \
-		libtool \
-		libtool-bin \
-		pkg-config \
-		software-properties-common \
-		texinfo \
-		udev \
-		cmake \
-		libglib2.0-dev
-
-ARG CLONE_DEFAULTS="--depth 5"
-
-# Use ToT alsa utils for the latest topology patches.
-RUN cd /tmp && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-lib.git && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-utils.git && \
-cd /tmp/alsa-lib && ./gitcompile &&  make install && \
-cd /tmp/alsa-utils && ./gitcompile &&  make install
-
-ARG GITHUB_SOF=https://github.com/thesofproject
-
-RUN cd $SRC && git clone $CLONE_DEFAULTS $GITHUB_SOF/sof
-WORKDIR sof
+# Base packages
+RUN apt-get -y update
+RUN apt-get install -y \
+      gettext git libc6-dev-i386 libglib2.0-dev libncurses5-dev \
+      libtool ninja-build python3-pip
+RUN pip3 install west
+
+# Zephyr SDK:
+#
+# Zephyr doesn't provide a clean "get the latest version" URL, but
+# note that the use of the /latest/ path component at least ensures
+# this will fail on a new release.
+ARG SDK_VER=0.16.1
+WORKDIR /root
+RUN curl -L https://github.com/zephyrproject-rtos/sdk-ng/releases/latest/download/zephyr-sdk-${SDK_VER}_linux-x86_64_minimal.tar.xz \
+    | tar xfJ -
+RUN zephyr-sdk-*/setup.sh -h
+
+WORKDIR $SRC
+
+# ALSA: (SOF needs tracks upstream closely, but doesn't yet pull these
+# via west or track commits internally)
+RUN git clone https://github.com/thesofproject/alsa-lib.git
+RUN git clone https://github.com/thesofproject/alsa-utils.git
+RUN cd $SRC/alsa-lib && ./gitcompile && make install
+RUN cd $SRC/alsa-utils && ./gitcompile && make install
```

Comment:
Dunno whether this matters here or not but we use a Docker image for that reason; to track ALSA: https://github.com/thesofproject/sof/blob/main/scripts/docker_build/sof_builder/Dockerfile

Reply:
Yeah, that's mostly a drive-by comment.  But it seems like there's space for SOF to specify "we are based on **this** version of alsa-lib/alsa-utils as of your current tree".  West would be handy, but then we'd have to rebuild alsa along with the firmware, and that's probably a pain.

Reply:
> Yeah, that's mostly a drive-by comment. But it seems like there's space for SOF to specify "we are based on this version of alsa-lib/alsa-utils as of your current tree". 

So maybe this comment is saying either too much or too little. Because the longer story is that only building topologies requires the latest and greatest ALSA, that's pretty much the only job the SOF "ALSA container" is used for. For building the kernel or the firmware a cutting edge ALSA has never been required in my experience.

Reply:
resolved